DigitalLearningSolutions.Data/Helpers/LearningHubSsoSecurityHelper.cs

```diff
@@ -0,0 +1,62 @@
+﻿namespace DigitalLearningSolutions.Data.Helpers
+{
+    using System;
+    using System.Security.Cryptography;
+    using System.Text;
+    using DigitalLearningSolutions.Data.Services;
+
+    public class LearningHubSsoSecurityHelper
+    {
+        private static readonly DateTime UnixEpoch = new DateTime(1970, 1, 1, 0, 0, 0, DateTimeKind.Utc);
+
+        private readonly IClockService clockService;
+
+        public LearningHubSsoSecurityHelper(IClockService clockService)
+        {
+            this.clockService = clockService;
+        }
+
+        public string GenerateHash(string state, string secretKey)
+        {
+            var secondsSinceEpoch = GetSecondsSinceEpoch();
+            var encoder = new UTF8Encoding();
+            var salt = encoder.GetBytes(secretKey);
+            var pwd = encoder.GetBytes(state + secondsSinceEpoch);

```

Comment:
Could the `pwd` var be renamed. Maybe `hashInput`? Feel free to come up with something better than that.